core/src/mindustry/logic/LExecutor.java

```diff
@@ -178,6 +179,23 @@ public Var(String name){
         void run(LExecutor exec);
     }
 
+    public static class LabelI implements LInstruction{
+        public int p1;
+
+        public LabelI(int p1){
+            this.p1 = p1;
+        }
+
+        @Override
+        public void run(LExecutor exec){
+            int icon = exec.obj(p1) instanceof UnlockableContent u ? u.iconId : 0;
+
+            if(exec.building(varThis) instanceof LogicBuild){
+                ((LogicBuild) exec.building(varThis)).label = icon;
+            }
```

Comment:
```suggestion
            if(exec.building(varThis) instanceof LogicBuild build){
                build.label = icon;
            }
```

Reply:
ah yes, instanceof magic, hard to get used to 👍